### PR TITLE
Logs

### PR DESCRIPTION
**Pull request by yermaalexx**
Add logging with logback to controllers, services, advices

### CODE REVIEW

File: .idea/compiler.xml
Comment (yermakovsa) [inline .idea/compiler.xml:1]: Please remove all .idea files from repository

File: src/main/java/com/alexx/employeesAndDepartments/EmployeesAndDepartmentsApplication.java [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
+@Slf4j
 public class EmployeesAndDepartmentsApplication {
-
     public static void main(String[] args) {
         SpringApplication.run(EmployeesAndDepartmentsApplication.class, args);
+        log.info("Starting application with {} args", args.length);
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/EmployeesAndDepartmentsApplication.java]: Why do you put args.length and not just args?

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java [hunk 2]
```diff
@@ -25,12 +27,14 @@ public DepartmentController(DepartmentService departmentService) {
     @PostMapping("/departments")
     @Operation(summary = "Add department", description = "Add new department, Request body with name of department required, ID is generated automatically")
     public ResponseEntity<DepartmentDTO> addDepartment(@RequestBody DepartmentDTO departmentDTO) {
+        log.info("Add department");
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java]: So if there is a problem with adding a department, who could we trace it in logs if you for every addDepartment put the same message? If there are millions of requests and you need to trace one request that failes

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java [hunk 3]
```diff
@@ -40,7 +44,9 @@ public ResponseEntity<Void> deleteDepartment(@PathVariable("departmentId")
                                                      @Min(value = 1, message = "ID of department must be positive.")
                                                      @Parameter(description = "ID of the existing department to remove")
                                                      Integer id) {
+        log.info("Delete department with ID: {}", id);
         departmentService.deleteDepartment(id);
+        log.info("Department deleted");
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java]: The same comment above applys to all logging

File: src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java [hunk 4]
```diff
@@ -50,6 +53,7 @@ public DepartmentDTO editDepartment(@Valid DepartmentDTO departmentDTO) {
             departmentEntity.setAdditionalInformation(departmentDTO.getAdditionalInformation());
         departmentRepository.save(departmentEntity);
         departmentDTO = modelMapper.map(departmentEntity, DepartmentDTO.class);
+        log.info("Department with ID {} edited", departmentDTO.getId());
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java]: Please read about different level of logging (DEBUG, INFO, TRACE) and apply what you feel is needed

File: src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java [hunk 5]
```diff
@@ -73,12 +77,14 @@ public DepartmentDTO getDepartment(Integer id) {
         DepartmentEntity departmentEntity = departmentRepository.findById(id).get();
         DepartmentDTO departmentDTO = modelMapper.map(departmentEntity, DepartmentDTO.class);
         departmentDTO.setListOfEmployees(employeeListOfDepartment(id));
+        log.info("Successfully");
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java]: Please read about best practices in logging and correct your messages

File: target/classes/application.properties
Comment (yermakovsa) [inline target/classes/application.properties:1]: Please remove target folder from repository

**Comment by yermakovsa** [comment]
> Add logging with logback to controllers, services, advices

Please put it in the name of pull request and please read best practices of creating names for pull requests

File: .idea/compiler.xml
Comment (yermaalexx) [reply .idea/compiler.xml:1]: I fix it in another pull request

File: src/main/java/com/alexx/employeesAndDepartments/EmployeesAndDepartmentsApplication.java (hunk 1 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/EmployeesAndDepartmentsApplication.java]: Added list of args in log

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java (hunk 2 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java]: I don't know if it's possible to store personal information (name, etc.) in the log

File: src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java (hunk 4 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java]: ok

File: src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java (hunk 5 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/services/DepartmentService.java]: ok, thank you

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java [hunk 6]
```diff
@@ -40,7 +46,10 @@ public ResponseEntity<Void> deleteDepartment(@PathVariable("departmentId")
                                                      @Min(value = 1, message = "ID of department must be positive.")
                                                      @Parameter(description = "ID of the existing department to remove")
                                                      Integer id) {
+        log.info("Delete department...");
+        log.debug("ID = {}", id);
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java:50]: Imagine you see this message in logs. The same message is written in editDepartnemt. log.debug("ID = {}", id). Withoud additional information this message is useless. Please correct it

File: src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java [hunk 7]
```diff
@@ -26,13 +28,17 @@ public EmployeeController(EmployeeService employeeService) {
     @Operation(summary = "Add employee", description = "Add new employee, Request body with name of employee required, " +
             "ID is generated automatically, birthDate must be over 18 years ago, idOfDepartment must match the existing department")
     public ResponseEntity<EmployeeDTO> addEmployee(@RequestBody EmployeeDTO employeeDTO) {
+        log.info("Add employee...");
+        log.debug("name: {}, birth date: {}", employeeDTO.getName(), employeeDTO.getBirthDate());
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java:32]: The same applys here as the message with ID

File: src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java [hunk 8]
```diff
@@ -26,13 +28,17 @@ public EmployeeController(EmployeeService employeeService) {
     @Operation(summary = "Add employee", description = "Add new employee, Request body with name of employee required, " +
             "ID is generated automatically, birthDate must be over 18 years ago, idOfDepartment must match the existing department")
     public ResponseEntity<EmployeeDTO> addEmployee(@RequestBody EmployeeDTO employeeDTO) {
+        log.info("Add employee...");
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java:31]: Please refer me a link where you read that you should do logging like this. And please describe in this comment best logging practices that you learnt

File: src/main/java/com/alexx/employeesAndDepartments/controllers/advices/ValidationErrorControllerAdvice.java [hunk 9]
```diff
@@ -26,6 +28,7 @@ public ValidationErrorResponse onConstraintValidationException(ConstraintViolati
                         violation.getMessage()
                 )
         ).collect(Collectors.toList());
+        log.warn(violations.toString());
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/controllers/advices/ValidationErrorControllerAdvice.java:31]: Please read again logging best practices. This time please look at examples. At many examples

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java (hunk 6 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java:50]: In log I will see something like this:

if ok:
2023-05-28 12:08:29,918 INFO Delete department...
2023-05-28 12:08:29,918 DEBUG ID = 1
2023-05-28 12:08:30,144 INFO Department with ID = 1 deleted

or if not:
2023-05-28 12:08:24,411 INFO Delete department...
2023-05-28 12:08:24,411 DEBUG ID = 2
2023-05-28 12:08:24,997 WARN Department with this ID does not exist

What is incorrect? 
Should I write in log all the DTO?

File: src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java (hunk 8 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java:31]: My logic is as follows:
first of all in INFO I write the essence of the called method (e.g. Add department),
then in DEBUG I write key parameters to be checked (name, ID, etc.),
if request completed, I write in INFO about it (e.g. New employee saved with ID = 1.)
if an exception has occurred, I write in WARN: (Name of employee must not be empty) or (Employee with this ID does not exist)

File: src/main/java/com/alexx/employeesAndDepartments/controllers/advices/ValidationErrorControllerAdvice.java (hunk 9 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/controllers/advices/ValidationErrorControllerAdvice.java:31]: Please, give me a link with best examples of logging in Rest api

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java (hunk 6 above)
Comment (yermakovsa) [reply src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java:50]: You should write more information. If there are mupltiple requests at the same time if will be hard to differentiate them, so you need to write more information

File: src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java (hunk 8 above)
Comment (yermakovsa) [reply src/main/java/com/alexx/employeesAndDepartments/controllers/EmployeeController.java:31]: Okay, sounds good. The only thing could you please remove dots from the end e.g. add employee...

**Pull request closed without merging**